backend/taco_backend.py

```diff
@@ -18,6 +18,43 @@
 import sparse
 from .core import Backend
 
+# Hack for @ operator for simplicity. No dimension checks.
+setattr(pt.tensor, '__matmul__', lambda x, y: TacoBackend.dot(x, y))
+
+
+def _scalarfunc(t):
+    if len(t.shape) == 0:
+        # TODO(yejiayu): Extend this to support any type casts.
+        return float(t[0])
+    else:
+        raise TypeError("Only rank 0 can be converted to scalars.")
+
+
+setattr(pt.tensor, '__float__', _scalarfunc)
+
+###############################################################################
+# Impl to get around pytaco slicing.
+###############################################################################
+__pt_impl_get_item = getattr(pt.tensor, '__getitem__')
+
+
+def _getitem(self, index):
+    if index is None or isinstance(index, int) or len(index) == 0:
```

Comment:
what does index represent when it's an int?

Reply:
AFAICT, index is just the arg passed into the bracket. 
E.g 

a = np.array([[0,0],[1,1]])
a[0] call will pass 0 as index into __getitem__

so the index can be any number/array/slices.

Reply:
So when the input is a list of ints, such as [0,1,2], we want to also just call pytaco indexing right? I guess the current implementation will Fallback to numpy slicing?

Reply:
I see what you mean, good point. 
Changed to type check slice().